### ISSUE TITLE
Validation crashes on top-level null-valued object attributes

### ISSUE THREAD

**Issue by DaineDanielson** [striglia/pyramid_swagger#197]
When validating an object like this:
```
{"title":"My Title","description":null}
```
Schema validation crashes with:
```
2017-03-22 17:07:29,672 ERROR [pyramid_debugtoolbar:233][waitress] Exception at http://localhost:6543/images/402e0db8-0de2-4c31-b423-6de2a302c58d
traceback url: http://localhost:6543/_debug_toolbar/exception?tb=4511419416&token=62275c7862665c783935653f725a5c7866365c786530325c78623627
Traceback (most recent call last):
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid_debugtoolbar/toolbar.py", line 215, in toolbar_tween
    response = _handler(request)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid_debugtoolbar/panels/performance.py", line 57, in resource_timer_handler
    result = handler(request)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid/tweens.py", line 51, in excview_tween
    request_iface=request_iface.combined
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid/view.py", line 612, in _call_view
    response = view_callable(context, request)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid/viewderivers.py", line 351, in authdebug_view
    return view(context, request)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid/viewderivers.py", line 409, in viewresult_to_response
    result = view(context, request)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid/tweens.py", line 22, in excview_tween
    response = handler(request)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid_swagger/tween.py", line 173, in validator_tween
    validation_context=validation_context)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid_swagger/tween.py", line 459, in _validate
    return f(*args, **kwargs)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/pyramid_swagger/tween.py", line 570, in swaggerize_request
    request_data = unmarshal_request(request, op)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/bravado_core/request.py", line 66, in unmarshal_request
    param_value = unmarshal_param(param, request)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/bravado_core/param.py", line 200, in unmarshal_param
    value = unmarshal_schema_object(swagger_spec, param_spec, raw_value)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/bravado_core/unmarshal.py", line 59, in unmarshal_schema_object
    return unmarshal_object(swagger_spec, schema_object_spec, value)
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/bravado_core/unmarshal.py", line 136, in unmarshal_object
    if schema.has_default(swagger_spec, prop_spec):
  File "/Users/daine/Developer/images/venv/lib/python3.5/site-packages/bravado_core/schema.py", line 22, in has_default
    return 'default' in swagger_spec.deref(schema_object_spec)
TypeError: argument of type 'NoneType' is not iterable
```

**Comment by macisamuele** [comment]
@DaineDanielson this issue is a bravado-core issue related to the last released version. More context is available on https://github.com/Yelp/bravado-core/issues/163.
Until the bravado-core issue is not fixed you'll continue to get this error.
For the moment my suggestion is to pin ``bravado-core < 4.7.0``

**Comment by DaineDanielson** [comment]
I was in the midst of writing my own PR for `bravado-core` when you replied with that link. Thanks!

**Comment by sjaensch** [comment]
Fixed in Yelp/bravado-core#165.
